### ISSUE TITLE
Shenmue - Doesn't save/load properly

### ISSUE THREAD

**Issue by GoogleCodeExporter** [aloone/nulldc#210]
```
What steps will reproduce the problem?

1: Start a new game, watch the intro
2: Choose the quick-save option (on Y button options), then reset game.
3: From title screen try to resume game.

What is the expected output? What do you see instead?
Game doesn't 'see' the save file, says there is no game to resume. 
The save file is there when checking via the Dreamcast boot menu.
Also, when starting a new game the game reckons the memory card is unformatted.

This only happens on the PAL version (I'm using GDI), saving on the NTSC 
version works fine. This is a shame as PAL versions of both Shenmues need to be 
played to carry over a save from 1 to 2.

What version of the product are you using? What build? What plugins?
r65, default plugins.

On what kind of system?
Operating System: Windows 7 x64
CPU: E8500 3.8Ghz
Video Card: Nvidia GTS 250 (ATI X1400 also tested)
Sound Card: Onboard Realtek ALC889A
```

Original issue reported on code.google.com by `matbur...@gmail.com` on 3 Sep 2010 at 12:58

**Comment by GoogleCodeExporter** [comment]
```
Are you sure that it's not a user error? Try again. Run the game, save, restart 
the emulator and run the game without changing any settings. Does it load?
```

Original comment by `Mr.PsyMan` on 3 Sep 2010 at 1:05

**Comment by GoogleCodeExporter** [comment]
```
I've just double-checked and it happens repeatedly. The issue doesn't happen if 
the load is attempted in the same emulator session (ie. save, reset to title 
screen (ABXY+Start), resume). I've attached some screenshots.

A quick Google shows it to be a longstanding problem:
http://forums.ngemu.com/nulldc-discussion/104078-shenmue-vmu-problem.html
http://www.game1010.com/video/video/AGTzlNUuirY/Shenmue-on-PC-NullDC-dreamcast-e
mulator.html (last comment)
```

Original comment by `matbur...@gmail.com` on 3 Sep 2010 at 9:52

Attachments:
- [Shenmue PAL save ok.jpg](https://storage.googleapis.com/google-code-attachments/nulldc/issue-210/comment-2/Shenmue PAL save ok.jpg)
- [Shenmue PAL can't resume.jpg](https://storage.googleapis.com/google-code-attachments/nulldc/issue-210/comment-2/Shenmue PAL can't resume.jpg)
- [Shenmue PAL VMU screen.jpg](https://storage.googleapis.com/google-code-attachments/nulldc/issue-210/comment-2/Shenmue PAL VMU screen.jpg)

**Comment by GoogleCodeExporter** [comment]
```
So, what are the steps to reproduce it ? from saving to being unable to load ..

Shenmue has various locks on the games, i always though this problem was 
because of that ....
```

Original comment by `drkIIRaziel` on 3 Sep 2010 at 10:18

**Comment by GoogleCodeExporter** [comment]
```
After saving, exit and restart nullDC, load Shenmue and choose 'Continue' from 
the main menu. Game will try to load the save point, but fail.
```

Original comment by `matbur...@gmail.com` on 3 Sep 2010 at 10:35

**Comment by GoogleCodeExporter** [comment]
```
Oh. it tries to load? Then indeed it is an issue. Nice find. :)
```

Original comment by `Mr.PsyMan` on 3 Sep 2010 at 3:41
- Changed title: **Shenmue - PAL version doesn't save/load properly**
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
i just found this page by googling the same problem. i think i've solved it. i 
just downloaded a pal bios and loaded from a resume save and from a bed save 
and they both work fine so far, i've played through a whole day and saved a few 
times, each time completely exiting nulldc then restarting the program and 
loading my saves, no problems. are you still using an ntsc bios perhaps?
```

Original comment by `js65...@gmail.com` on 15 Sep 2010 at 11:30

**Comment by GoogleCodeExporter** [comment]
```
That's interesting, yes I do use an NTSC bios with 'Patch GDROM Region' enabled.
```

Original comment by `matbur...@gmail.com` on 16 Sep 2010 at 9:33

**Comment by GoogleCodeExporter** [comment]
Original comment by `KrossX3` on 17 Sep 2010 at 12:15
- Added labels: **Type-Defect**
- Removed labels: **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
```
So, is this a bug or not ?
```

Original comment by `drkIIRaziel` on 27 Sep 2010 at 9:50

**Comment by GoogleCodeExporter** [comment]
```
  I know one thing about Shenmue! 
  When we close the game with emulator  we must sure what its already saved! Only after that we can close the NullDc. If we will do it earlier than it must be the savefile will be broken! It happend to me many times, but, ofcourse if I doing all correctly all works fine. 

  So, I think it's not a bug, It's just game's specification. 

  P.s.
  But maybe i'm wrong, who knows?...
```

Original comment by `N2008eo...@rambler.ru` on 27 Sep 2010 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
Yes, this is a bug. I've played the PAL version on proper NTSC hardware which 
saves fine.
```

Original comment by `matbur...@gmail.com` on 1 Oct 2010 at 6:56

**Comment by GoogleCodeExporter** [comment]
```
I still don't think that this is a bug. From what I saw I'm quite convinced 
that you're using a hacked BIOS which leads to this issue.
```

Original comment by `Mr.PsyMan` on 1 Oct 2010 at 7:38

**Comment by GoogleCodeExporter** [comment]
```
A hacked BIOS? :D

I'm using a TOSEC verified US BIOS. Why not try to recreate yourself so we can 
be sure?
```

Original comment by `matbur...@gmail.com` on 1 Oct 2010 at 11:54

**Comment by GoogleCodeExporter** [comment]
```
When I said hacked BIOS I meant the flash ROM, not the boot ROM. Not really 
willing check it out myself, maybe you should ask raz.
```

Original comment by `Mr.PsyMan` on 1 Oct 2010 at 12:18

**Comment by GoogleCodeExporter** [comment]
```
The flash ROM comes with the TOSEC boot ROM, so it should be good.

I guess js65439 had the same problem, as he found this issue by googling for a 
solution, so that's 2 users with the same issue.

I do appreciate this one sounds like dodgy files, but I've used TOSEC files 
throughout (BIOS & GDI).
```

Original comment by `matbur...@gmail.com` on 1 Oct 2010 at 6:36

**Comment by GoogleCodeExporter** [comment]
```
On normal hardware, you won't be able to play PAL games on NTSC, or the other 
way around ...

This makes me think i should remove the patch gdrom region option, seems to be 
the cause for this report :p
```

Original comment by `drkIIRaziel` on 2 Oct 2010 at 12:59

**Comment by GoogleCodeExporter** [comment]
```
Should have mentioned I have a chipped NTSC Dreamcast to play imports, there's 
also the DC-X option. Not sure how much these affect the BIOS with regards to 
this issue.

Please don't remove the patch gdrom option, that would make playing imports 
really awkward. Particularly if only this game has an issue with it, I'd rather 
switch to a PAL BIOS for this game only than switch the BIOS for every import.


```

Original comment by `matbur...@gmail.com` on 2 Oct 2010 at 2:36

**Comment by GoogleCodeExporter** [comment]
```
PACTH GDROM IS WONDERFUL OPTION!!!!!!!!!!!!!
DO NOT DELETE THIS! NO WAY! PLEASE!!!!!!!! 

Shenmue works fine for me!!!  I finished Shenmue I & II withot problems. I have 
PAL & NTSC. Both works fine! Both is playable. Saving without problem!  I've 
never have this issue! When I play Shenmue I have only issue 150 and nothing 
more! Everything else is normal, how it possible now!

```

Original comment by `N2008eo...@rambler.ru` on 2 Oct 2010 at 9:11

**Comment by GoogleCodeExporter** [comment]
```
N2008 - you can save PAL Shenmue I when using an NTSC BIOS? I know the PAL 
version will save with a PAL BIOS, and likewise NTSC with NTSC.

I'm unsure whether this is an issue now, seeing as playing the PAL version on a 
vanilla NTSC Dreamcast wouldn't work anyway. I guess it depends on what a 
chipped Dreamcast and/or the DC-X disc do differently, and how that compares to 
what the 'Patch GDROM Region' does. I know for definite though that PAL Shenmue 
can be played on a chipped NTSC Dreamcast, and saving works.
```

Original comment by `matbur...@gmail.com` on 2 Oct 2010 at 10:19

**Comment by GoogleCodeExporter** [comment]
```
Can you post a screenshot from when the emulator boots the NTSC BIOS?
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 9:14

**Comment by GoogleCodeExporter** [comment]
```
Did you want a shot of the trace window, or the output window?
Here's a shot of the output window, it's an NTSC-red swirl rather than 
PAL-blue. Let me know if you need more info.
```

Original comment by `matbur...@gmail.com` on 4 Oct 2010 at 11:15

**Comment by GoogleCodeExporter** [comment]
```
Not good. I want the whole window.
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 11:25

**Comment by GoogleCodeExporter** [comment]
Original comment by `matbur...@gmail.com` on 4 Oct 2010 at 11:39

Attachments:
- [DC NTSC boot.jpg](https://storage.googleapis.com/google-code-attachments/nulldc/issue-210/comment-23/DC NTSC boot.jpg)

**Comment by GoogleCodeExporter** [comment]
```
I see...
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 12:04

**Comment by GoogleCodeExporter** [comment]
```
I've retested using a PAL BIOS and the same problem occurs - if I save using 
the 'resume' option in-game (confusing, yes), then exit and restart nullDC and 
use the 'continue' option, the load is unsuccessful. Also, trying to start a 
new game gives a 'memory card unformatted' error, which isn't true according to 
the DC VMU manager or other games, which save/load fine.

Only happens with the PAL version, the NTSC version saves/loads fine.

N2008, what is this 'correct sequence' for saving that you mention in comment 
10?
```

Original comment by `matbur...@gmail.com` on 4 Oct 2010 at 12:04

**Comment by GoogleCodeExporter** [comment]
```
'I see...' :D

It's a bit strange that some people here claim PAL Shenmue saves/loads fine.
```

Original comment by `matbur...@gmail.com` on 4 Oct 2010 at 12:13

**Comment by GoogleCodeExporter** [comment]
```
It does. But only at 50Hz. As long as the initial save is created at 50Hz mode 
it will load fine. But for some reason the load fails if the game was saved at 
60Hz mode.

Will investigate when feeling like it.
```

Original comment by `Mr.PsyMan` on 4 Oct 2010 at 1:56

**Comment by GoogleCodeExporter** [comment]
```
That's weird, thanks for the workaround.
```

Original comment by `matbur...@gmail.com` on 4 Oct 2010 at 2:16

**Comment by GoogleCodeExporter** [comment]
```
OK, looking at it. It's an interesting and rare yet trivial issue that makes no 
sense.
My favorite.
```

Original comment by `Mr.PsyMan` on 22 Oct 2010 at 1:26
- Changed state: **Started**
- Added labels: **Priority-Medium**
- Removed labels: **Priority-Low**

**Comment by GoogleCodeExporter** [comment]
```
So, after tracking the issue it turns out that it is not so rare, nor trivial 
after all.

Great. Must talk to raz about this as it affects quite some games, in a 
different way.

If you are encountering this bug and you're desperate you can avoid it by 
starting the emulator without any VMUs connected. Then load the VMU with the 
save when you reach the main menu of the game.
VMUs are hot-swappable so you can (dis)connect them while the emulator runs 
from the options->maple menu.
```

Original comment by `Mr.PsyMan` on 23 Oct 2010 at 4:09
- Added labels: **emu-dma**

**Comment by GoogleCodeExporter** [comment]
```
I can verify that this works if the maple dma is delayed by 16 ms. No idea what 
is the -actual- cause behind it though, too lazy to track it right now.
```

Original comment by `drkIIRaziel` on 23 Oct 2010 at 3:00

**Comment by GoogleCodeExporter** [comment]
```
I told you,  will do it tonight raz. Go sleep or something. :p
```

Original comment by `Mr.PsyMan` on 23 Oct 2010 at 3:30

**Comment by GoogleCodeExporter** [comment]
```
Should be fixed in r84, but won't close this as further investigation needs to 
be done ...
```

Original comment by `drkIIRaziel` on 25 Oct 2010 at 5:28

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Hello, just to say to you guys, I have the same problem, I can't help you with 
anything but just telling this :

You cannot make more than ONE save InGame, if you save at the bed and want then 
to "resume" it will say to you that your VMU is corrupted and needs to be 
formated (even if you've done it before) so what I tried is :

Format the VMU by opening another instance of nulldc ---> Doesn't work.

Create a new VMU that I also formated and DELETE the first one then renaming 
the "vmu_data_port02.bin" into "vmu_data_port01.bin" ----> Doesn't work either.

Tried to format the VMU with "vmuexplorer" -> doesn't change anything

Tried to save on the 2nd vmu while it was plugged into the 2nd slot --> shenmue 
won't allow it.

But it does recognize if you unplug the VMU and says to you that no memory card 
is inserted

So the game is totally unplayable since you cannot do it without saving.
Hope this help.


EDIT 1H30 LATER : Seems you can fix the problem by just setting 
"Cable Type" to "TV (VBS/Y+S/C)"  
"System Region" to "Eur"
"Broadcast Format" to "Pal"

Saves and loads will work great now. The only thing is that you are forced to 
play in 50 FPS due to 50Hertz Pal mode.
I wish there would be a "save state" future into the emulator or a way to 
control the framerate to push it to 60 even in PAL...
```

Original comment by `siriu...@gmail.com` on 28 Oct 2010 at 6:02

**Comment by GoogleCodeExporter** [comment]
```
Or maybe not. This just happened to me with the pal bios. So this shit is 
freaking annoying !
```

Original comment by `siriu...@gmail.com` on 31 Oct 2010 at 2:26

**Comment by GoogleCodeExporter** [comment]
```
The particular problem was fixed on revision 84... just take some time to 
actually read before posting.
```

Original comment by `Mr.PsyMan` on 31 Oct 2010 at 3:47

**Comment by GoogleCodeExporter** [comment]
```
 I use r84. Shenmue hangs on game start, when creating save file for the game. Pal version.
```

Original comment by `x.ran...@gmail.com` on 3 Nov 2010 at 5:46

**Comment by GoogleCodeExporter** [comment]
```
 Tested with 1.03 - no such problem.
```

Original comment by `x.ran...@gmail.com` on 3 Nov 2010 at 5:53

**Comment by GoogleCodeExporter** [comment]
```
Issue 329 has been merged into this issue.
```

Original comment by `KrossX3` on 29 Nov 2010 at 3:56

**Comment by GoogleCodeExporter** [comment]
```
Happens with NTSC-J too now.
```

Original comment by `KrossX3` on 29 Nov 2010 at 4:17
- Changed title: **Shenmue - Doesn't save/load properly**

**Comment by GoogleCodeExporter** [comment]
```
Reverting r84 makes it work again. Patch attached.
```

Original comment by `KrossX3` on 29 Nov 2010 at 4:57

Attachments:
- [Revert r84-ish.patch](https://storage.googleapis.com/google-code-attachments/nulldc/issue-210/comment-42/Revert r84-ish.patch)

**Comment by GoogleCodeExporter** [comment]
```
hello, what i have to do whit this r84 file?
```

Original comment by `CrazyRap...@googlemail.com` on 13 Jan 2012 at 3:13

**Comment by GoogleCodeExporter** [comment]
```
I had the same problem, but it just happened in cd 3 after i got a job in the 
harbor. Really weird

```

Original comment by `davidder...@gmail.com` on 5 Feb 2012 at 7:52

**Comment by GoogleCodeExporter** [comment]
```
Saving issue is very simple to solve if using the PAL version:

Ensure Patch GDRom Region is enabled
Go to Options > NullDC Settings
System region: EUR
Broadcast region: PAL

I had the formatting error at first but it now saves just fine.  Hope this 
helps someone.
```

Original comment by `weedfee...@gmail.com` on 19 Mar 2012 at 5:20

**Comment by GoogleCodeExporter** [comment]
```
wierd semi-related issue here. When loading form a resume file in a new nulldc 
session, sometimes I can no longer save - says "Memory card is unformatted" 
when I go to do either a resume save or a save in Ryo's bed. Really 
frustrating, as I am losing a bunch of progress every now and again, seemingly 
at random. Issue is fixed vy restarting the emulator, but obviously that is a 
non-solution given that I need to be in the same session to save my progress...

Using Shenmue US images (NTSCJ) with USA Bios and NTSC video.
```

Original comment by `clockwor...@gmail.com` on 16 Dec 2012 at 9:50

**Comment by GoogleCodeExporter** [comment]
```
Well, i want to share this info. at this point:

I'm using NullDC r149 and with Shenmue (Part 1), I run the game with the 
Settings: 

System Region: USA
Broadcast Region: NTSC

And i have notice this: When i run the PAl GDI complete file of Shenmue it asks 
for a 50 or 60 Htz. option, right? Then i chose 60 the first try. The game runs 
BUT, as stated before, when you save, and some other day want to continue, the 
game can not load the save file!

So what i did is to erase that file save and then when it asks for the Htz. 
option go for 50 Htz. and even with the settings described above, the game 
saves and loads with no issues. 

The important thing is to choose 50 Htz. so the game can be played and finished!
```

Original comment by `nanke...@gmail.com` on 13 Mar 2013 at 8:04

**Comment by GoogleCodeExporter** [comment]
```
Oh! And in this revision, it's not possible to play it at 60 Htz. But i hope 
somebody could tell me the difference, 'cause i can't see any between 50 or 60 
in the emulator!
```

Original comment by `nanke...@gmail.com` on 13 Mar 2013 at 8:07

**Comment by GoogleCodeExporter** [comment]
```
Hello all, I am new here but I had same problem, I managed to fix it so I am 
letting you all know.

System region set to EUR
Broadcast format set PAL
And Cable type set VGA(0)

it fixes it although, it adds a little lag to the game. Hope it helped someone

edit: set those things in the emulator 
```

Original comment by `Arvar...@gmail.com` on 16 Aug 2013 at 3:16

**Comment by GoogleCodeExporter** [comment]
```
That worked for me - with my loading issue: 

Ensure Patch GDRom Region is enabled
Go to Options > NullDC Settings
System region: EUR
Broadcast region: PAL
Cable type set VGA(0)

THANKS!!!
```

Original comment by `kjakubow...@gmail.com` on 13 Apr 2014 at 11:14

**Comment by GoogleCodeExporter** [comment]
```
hi guys i have an issue i had unistalled a custom bios from my flash chip and i 
can´t boot the console anymore ,it gives me the red swirl on my pal console 
but it doesn´t boot sd loader ,can i make a bootable cd to reflash my bios 
again ,since it´s not piggyback?

Best Regards

JOSÉ
```

Original comment by `JRDG1...@gmail.com` on 23 Feb 2015 at 5:34
